brave/brave-core
# feat(wallet): iOS Wallet Network Settings

Resolves https://github.com/brave/brave-browser/issues/37741
The `Show Test Networks` toggle is now removed
All networks including primary networks, sub-networks, test networks, custom neworks, are all listed in this screen



## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues/37741) for my issue
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
- network lists 
1. in an older version of nightly Brave, make sure `Show Test Networks` is turned off
2. install a nightly with this PR in.
3. open wallet and go to wallet settings / networks
4. you should see test networks for each coin type are listed with the `eye-closed` icon indicates they are hidden 
5. Repeat step 1 to 4, but turn on `Show Test Networks`. You should see test networks for each coin type are listed with `eye-opened` icon indicates they are not hidden.
- default networks
1. Go to wallet settings/networks
2. For each coin type, there is one network with `eye-opened` icon but grey-out indicates hiding this network is disabled. This also means this is the default network for this coin type. 
3. tap the 3-dot button, only `Edit` is available 
- add custom networks 
1. try to add a new custom network
2. it will listed in the network list.
3. tap the 3-dot button, you can `Edit`/`Delete`/`Set as Default`
- remove custom networks
1. Remove a custom network via the 3-dot context menu
2. This network should be removed from the list
- edit networks 
1. Edit a network via the 3-dot context menu
2. save the change should be successfully 
- Hide networks
1. hide some networks by tapping the `eye-opened` icon
2. it will become `eye-closed` icon if its hidden
3. after that, check hidden networks should not be listed in filter or network selection screen. Assets that are grouped in hidden networks should not be displayed 
- unhide networks
1. unhide some networks by tapping the `eye-closed` icon
2. it will become `eye-opened` icon if its not hidden
3. after that, check hidden networks should be listed in filter or network selection screen. Assets that are grouped in these networks should be displayed
